user_guide_src/add-edit-this-page

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env php
+<?php declare(strict_types=1);
+
+if ($argc === 1) {
+    echo 'Usage: ' . $argv[0] . ' <html-dir>' . PHP_EOL;
+    echo '  e.g: ' . $argv[0] . ' user_guide_src/build/html/' . PHP_EOL;
+
+    exit(1);
+}
+
+$dir = realpath($argv[1]);
+
+if ($dir === false) {
+    throw new RuntimeException('Cannot find directory: ' . $dir);
+}
+
+$iterator = new RecursiveIteratorIterator(
+    new RecursiveDirectoryIterator(
+        $dir,
+        FilesystemIterator::CURRENT_AS_FILEINFO |
+        FilesystemIterator::KEY_AS_PATHNAME |
+        FilesystemIterator::SKIP_DOTS
+    )
+);
+
+$files = new RegexIterator(
+    $iterator,
+    '/\A.+\.html\z/i',
+    RecursiveRegexIterator::MATCH
+);
+
+foreach ($files as $file_path => $file_info) {
+    echo 'processing... ' . $file_path . PHP_EOL;
+
+    $uriPath    = str_replace('.html', '.rst', str_replace($dir, '', $file_path));
+    $gitHubLink = '<a class="btn btn-neutral float-right" href="https://github.com/codeigniter4/CodeIgniter4/edit/develop/user_guide_src/source'
+        . $uriPath . '">Edit this page</a>';
+
+    $content = file_get_contents($file_path);
+
+    $pattern = '!">Edit this page</a>!u';
+    if (preg_match($pattern, $content) === 1) {
+        echo 'skip' . PHP_EOL;
```

Comment:
include the file path of what is being skipped

Reply:
Do you really need it?

```
processing... /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/filesystem_helper.html
skip: /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/filesystem_helper.html
processing... /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/number_helper.html
skip: /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/number_helper.html
processing... /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/url_helper.html
skip: /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/url_helper.html
```
```
processing... /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/filesystem_helper.html
skip
processing... /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/number_helper.html
skip
processing... /Users/kenji/work/codeigniter/CodeIgniter4/user_guide_src/build/html/helpers/url_helper.html
skip
```

Reply:
Yeah, much better